indexer/services/roundtable/src/helpers/aws.ts

```diff
@@ -33,13 +42,75 @@ export async function getMostRecentDBSnapshotIdentifier(rds: RDS): Promise<strin
     throw Error(`No DB snapshots found with identifier: ${config.RDS_INSTANCE_NAME}`);
   }
 
+  let snapshots: RDS.DBSnapshotList = awsResponse.DBSnapshots;
+  // Only include snapshots with snapshot identifier that starts with prefixInclude
+  if (snapshotIdentifierPrefixInclude !== undefined) {
+    snapshots = snapshots
+      .filter((snapshot) => snapshot.DBSnapshotIdentifier &&
+        snapshot.DBSnapshotIdentifier.startsWith(snapshotIdentifierPrefixInclude),
+      );
+  }
+  if (snapshotIdentifierPrefixExclude !== undefined) {
+    snapshots = snapshots
+      .filter((snapshot) => snapshot.DBSnapshotIdentifier &&
+        !snapshot.DBSnapshotIdentifier.startsWith(snapshotIdentifierPrefixExclude),
+      );
+  }
+
   logger.info({
     at: `${atStart}getMostRecentDBSnapshotIdentifier`,
     message: 'Described snapshots for database',
-    mostRecentSnapshot: awsResponse.DBSnapshots[awsResponse.DBSnapshots.length - 1],
+    mostRecentSnapshot: snapshots[snapshots.length - 1],
   });
 
-  return awsResponse.DBSnapshots[awsResponse.DBSnapshots.length - 1].DBSnapshotIdentifier!;
+  return snapshots[snapshots.length - 1]?.DBSnapshotIdentifier;
+}
+
+/**
+ * @description Create DB snapshot for an RDS database. Only returns when the
+ * snapshot is available.
+ */
+export async function createDBSnapshot(
+  rds: RDS,
+  snapshotIdentifier: string,
+  dbInstanceIdentifier: string,
+): Promise<string> {
+  const params = {
+    DBInstanceIdentifier: dbInstanceIdentifier,
+    DBSnapshotIdentifier: snapshotIdentifier,
+  };
+
+  try {
+    await rds.createDBSnapshot(params).promise();
+    // Polling function to check snapshot status. Only return when the snapshot is available.
+    const waitForSnapshot = async () => {
```

Comment:
nit: I think you can use this `waiter` method/object that is part of the AWS sdk [here](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/RDS.html#dBSnapshotAvailable-waiter) rather than writing your own. Non-blocking though, if it's too complicated to use, no need to use it.

Reply:
done